### PR TITLE
feat: show numbers of regression and error suites in generated tests

### PR DESCRIPTION
**Pull request by antipeon**

### CODE REVIEW

**Comment by antipeon** [comment]
Screenshot (23)

**Comment by antipeon** [comment]
first
second
third

File: vscode-plugin/src/logger.ts [hunk 1]
```diff
@@ -84,11 +85,29 @@ export class Logger {
...
                     }
+                },
+                vscodeOutChannelForClientOnlyAppender : {
+                    type: vscodeOutChannelAppender,
+                    layout: {
+                        type: 'pattern',
+                        pattern: this.infoPatternString
+                    }
+                },
+                vscodeOutChannelForMoreThanDebug: {
```
Comment (ladisgin) [inline vscode-plugin/src/logger.ts:96]: rename

File: vscode-plugin/src/responses/responseHandler.ts [hunk 2]
```diff
@@ -51,7 +51,15 @@ export class TestsResponseHandler implements ResponseHandler<TestsResponse> {
             await Promise.all((testsFiles).map(async (test) => {
                 const localPath = pathUtils.substituteLocalPath(test.getFilepath());
                 const testfile = vs.Uri.file(localPath);
-                logger.info(`Write test file ${test.getFilepath()} to ${localPath}`);
+                
+                if (testfile.path.endsWith('_test.cpp')) {
+                    const codeText = test.getCode();
+                    const testsNumberInErrorSuite = (codeText.match(/TEST\(error/g) || []).length;
```
Comment (ladisgin) [inline vscode-plugin/src/responses/responseHandler.ts:57]: Can count incorrect, if has "/TEST\(error/g)" into test body.

**Review by ladisgin — CHANGES_REQUESTED** [review]

File: server/src/printers/TestsPrinter.cpp [hunk 3]
```diff
@@ -55,26 +55,31 @@ void TestsPrinter::joinToFinalCode(Tests &tests, const fs::path& generatedHeader
     }
     writeStubsForStructureFields(tests);
     ss << NL;
-    printSuite(Tests::DEFAULT_SUITE_NAME, tests.methods);
-    printSuite(Tests::ERROR_SUITE_NAME, tests.methods);
+    uint32_t regressionMethodsNumber = printSuiteAndReturnMethodsCount(Tests::DEFAULT_SUITE_NAME, tests.methods);
```
Comment (ladisgin) [inline server/src/printers/TestsPrinter.cpp:58]: unused variables

**Review by ladisgin — CHANGES_REQUESTED** [review]

**Review by ladisgin — APPROVED** [review]

**Pull request merged**